### DISCUSSION TITLE
Issue with Missing "sync.py" File in pymodbus

### DISCUSSION THREAD

**Discussion by Felix1978-ab** [pymodbus-dev/pymodbus#1727] [Help] [answered]
Description:
I have been trying to run a Python script that uses the "pymodbus" package to communicate with a Modbus device. Despite multiple attempts to install the necessary packages, the "sync.py" file required by the script is still missing. The script should connect to a Modbus slave and read a holding register, but it cannot proceed due to this missing file.

I followed the installation steps, verified that the required packages are installed, and ensured that the Modbus device is reachable. However, I still encounter the issue of the missing "sync.py" file.

Code Sample (without sensitive information):

python
Copy code
from pymodbus.client.sync import ModbusTcpClient

def main():
    # Modbus Slave IP address (replace with actual IP)
    modbus_slave_ip = "xxx.xxx.xxx.xxx"
    # Modbus TCP Port (replace with actual port)
    modbus_slave_port = 502

    try:
        # Establish connection to Modbus slave
        client = ModbusTcpClient(modbus_slave_ip, modbus_slave_port)
        client.connect()

        # Read a Holding Register (Address 0) from the Modbus slave
        response = client.read_holding_registers(0, 1, unit=0x01)
        if response.isError():
            print(f"Error reading register: {response}")
        else:
            value = response.registers[0]
            print(f"Register value: {value}")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        # Close the connection
        client.close()

if __name__ == "__main__":
    main()

Any assistance in resolving this issue would be greatly appreciated. Thank you in advance for your help!

**Comment by janiversen** [comment, marked as answer]
we have not had a sync.py for a very long time, I suggest you read our documentation on how to use the version of pymodbus you are using.

And btw. your read is also wrong.

It is of course your choice, but you might be better served, by using our examples, which are tested against every version. simple_client_sync.py seems to do what you need.

**Reply by Felix1978-ab** [reply]
thanks, i will look the doku :-)
